doc/source/reference/arrays.scalars.rst

```diff
@@ -169,6 +174,25 @@ Inexact types
    :members: __init__
    :exclude-members: __init__
 
+.. note::
+
+   The printing behavior of inexact scalars is to use as few digits as possible
+   to display the result unambiguously. This means that equal values at
+   different precisions may display differently::
```

Comment:
Look good @ahaldane?
```suggestion
   Inexact scalars are printed using the fewest decimal digits needed to
   uniquely identify their value from other values of the same datatype,
   by judicious rounding. See the ``unique`` parameter of
   `format_float_positional` and `format_float_scientific`.

   This means that variables with equal binary values but whose datatypes are of
   different precisions may display differently::
```

I don't think `set_printoptions` has anything useful to add here.